### PR TITLE
luci-app-speedtest: Add packages

### PR DESCRIPTION
**Pull request by animegasan**
# LuCI for Speedtest

image

I feel this is very helpful for those of us who don't want the hassle of going to the terminal just to check the internet speed of OpenWrt devices. Especially for those who are remote accessing from outside the network.

#6949

### CODE REVIEW

**Comment by dannil** [comment]
Please don't close you PR:s as soon as you make changes as that makes all the comments get lost, you can just continue pushing to the same branch and your current PR will get updated automatically.

**Comment by hnyman** [comment]
You can also squash (= merge together) the existing commits to clean up things.
Please read advice in:
https://github.com/openwrt/luci/blob/master/CONTRIBUTING.md#advice-on-pull-requests

**Comment by systemcrash** [comment]
What the others are saying @animegasan is to go practice force pushing and adding commits to a branch. Do this before ooening a pr and observe what happens in your GitHub repo when making a force push despite any scary warnings that might appear in your client telling you not to do that. 

Practice.

File: applications/luci-app-speedtest/root/usr/share/rpcd/acl.d/luci-app-speedtest.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+	"luci-app-speedtest": {
+		"description": "Grant access to speedtest",
+		"write": {
```
Comment (systemcrash) [inline applications/luci-app-speedtest/root/usr/share/rpcd/acl.d/luci-app-speedtest.json:4]: Is it a problem to use `		"read": {` instead?

File: applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+						var value = dropdown.options[dropdown.selectedIndex].value;
+						var command = ['--output', 'json'];
+						if (value !== 'all') {
+							command.push('--' + value);
+						}
+
+						var status = document.getElementById('speedtest-status');
+						status.textContent = _('Running');
+
+						fs.exec_direct('speedtest', command)
```
Comment (systemcrash) [inline applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js:97]: So the app assumes that the speedtest binary is available. Let's say that the binary was deleted, or some sysupgrade went bad (even tho this app depends on it), what happens when the binary is absent? Is the error message misleading or must we perform a separate check at init?

**Review by systemcrash — APPROVED** [review]

**Comment by animegasan** [comment]
> Please don't close you PR:s as soon as you make changes as that makes all the comments get lost, you can just continue pushing to the same branch and your current PR will get updated automatically. You should probably close this and re-use #6949 where the comments are.

I appreciate your feedback. I'd like to explain that I couldn't reopen the previous PR because I had already deleted the associated branch. As I'm relatively new to using GitHub, I'm still learning about best practices. However, I truly value your advice not to close PRs prematurely and will strive to continue learning and improving. Thank you for your understanding and support.

> You can also squash (= merge together) the existing commits to clean up things. Please read advice in: https://github.com/openwrt/luci/blob/master/CONTRIBUTING.md#advice-on-pull-requests

I understand, and I admit that I've often struggled with this. I recognize that my previous PRs may have seemed spammy. This time around, I'm making a conscious effort not to repeat the same mistakes. Thank you for your patience and understanding

> What the others are saying @animegasan is to go practice force pushing and adding commits to a branch. Do this before opening a pr and observe what happens in your GitHub repo when making a force push despite any scary warnings that might appear in your client telling you not to do that.
> 
> Practice.

Yes, I spent the whole night practicing on another repository, and I finally understand how to maintain my PRs according to the rules. Thank you.

File: applications/luci-app-speedtest/root/usr/share/rpcd/acl.d/luci-app-speedtest.json (hunk 1 above)
Comment (animegasan) [reply applications/luci-app-speedtest/root/usr/share/rpcd/acl.d/luci-app-speedtest.json:4]: I haven't tried it yet, but I'll give it a shot. Perhaps it'll work out.

File: applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js (hunk 2 above)
Comment (animegasan) [reply applications/luci-app-speedtest/htdocs/luci-static/resources/view/speedtest.js:97]: The action has been handled here.
```js
.catch(function(err) {
	if (!err.response) {
		status.textContent = _('No response received, please check your internet connection or try again later.');
	} else if (err.response.data && err.response.data.error === 'unable to retrieve your ip info') {
		status.textContent = _('Unable to retrieve IP information, please check your internet connection.');
	} else {
		status.textContent = err;
	}
	running = false;
});
```

**Comment by animegasan** [comment]
Maybe this package can be forwarded by @systemcrash . 

I have to learn again how to contribute. Thank You

**Comment by dannil** [comment]
> Maybe this package can be forwarded by @systemcrash .
> 
> Look how I messed it up again
> 
> image
> 
> I have to learn again how to contribute. Thank You

No worries, we are all beginners at some point in time! As long as your PR isn't merged you're free to force push to make the code/commits look correct. 

In your case I would reset the merge, run `git fetch` to get the latest from origin (probably your repo if you haven't done any custom setup with the remotes) and run `git rebase origin/master` (remember to sync the changes to your fork's `master` from this repo's `master` otherwise you won't get new commits to pull down). I'm not personally a fan of `git merge` as it introduces merge commits. Also, the reflog contains all your historical commands, so even if you delete a branch you can get the content back.

**Comment by systemcrash** [comment]
Yeah no worries. I had to learn the ropes also. 

Also no merge commits. Read up on how to do a rebase of your branch on master. It will be problem free since this is a new app.

**Comment by systemcrash** [comment]
ping @animegasan

**Comment by castillofrancodamian** [comment]
A speedtest app like OpenSpeedTest wouldn't hurt.

**Review by dannil — CHANGES_REQUESTED** [review]
Make sure to clean up the commits.

**Pull request closed without merging**